backend/src/test/java/com/google/models/AnomalyTest.java

```diff
@@ -120,12 +121,19 @@ public void toString_correctConversion() {
     assertEquals(expectedStr.toString(), ANOMALY.toString());
   }
 
+  @SuppressWarnings("unchecked")
   @Test
   public void toEntity_correctAnomalyToEntityConversion() {
     Entity anomalyEntity = ANOMALY.toEntity();
     EmbeddedEntity dataPointsEE = 
         (EmbeddedEntity) anomalyEntity.getProperty(Anomaly.DATA_POINTS_PROPERTY);
-    
+    List<EmbeddedEntity> relatedDataEEList = 
+        (List<EmbeddedEntity>) anomalyEntity.getProperty(Anomaly.RELATED_DATA_LIST_PROPERTY);
+    List<RelatedData> relatedDataList = relatedDataEEList.stream()
+        .map(RelatedData::createFromEmbeddedEntity)
+        .collect(ImmutableList.toImmutableList());
```

Comment:
ooc, why do you have to convert the embedded entities into objects for the related data but not for the data points?

Reply:
this is because data points is a map between timestamps and values, so I can compare the timestamps and values directly by getting the properties from data points embedded entities. I do this in a helper function: `boolean embeddedEntityDataPoints_equal(...)`.
`relatedDataList` is stored as a list of embedded entities, so in order to compare the embedded entities to see if they are the same as a `RelatedData` object, I convert them back into objects.
hopefully this makes sense (?)

Reply:
yup that makes sense, thanks!